### ISSUE TITLE
Concat no longer supports Series containing Null values

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#452]
`Concat` method no longer supports concatenation of Series containing Nulls. Previously, `concat` supported nulls but `append` did not.

Examples below:

```
import cudf

s = cudf.Series([1,2,3,None,4])
print(cudf.concat([s, s]))

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/conda/envs/cudf/lib/python3.5/site-packages/cudf/dataframe/columnops.py in as_column(arbitrary, nan_as_null)
    297         try:
--> 298             data = as_column(memoryview(arbitrary))
    299         except TypeError:

TypeError: memoryview: a bytes-like object is required, not 'list'

During handling of the above exception, another exception occurred:

CudaAPIError                              Traceback (most recent call last)
<ipython-input-19-de412e0728e2> in <module>()
----> 1 s = cudf.Series([1,2,3,None,4])
      2 print(cudf.concat([s, s]))

/conda/envs/cudf/lib/python3.5/site-packages/cudf/dataframe/series.py in __init__(self, data, index, name, nan_as_null)
     73 
     74         if not isinstance(data, columnops.TypedColumnBase):
---> 75             data = columnops.as_column(data, nan_as_null=nan_as_null)
     76 
     77         if index is not None and not isinstance(index, Index):

/conda/envs/cudf/lib/python3.5/site-packages/cudf/dataframe/columnops.py in as_column(arbitrary, nan_as_null)
    298             data = as_column(memoryview(arbitrary))
    299         except TypeError:
--> 300             data = as_column(pa.array(arbitrary))
    301 
    302     return data

/conda/envs/cudf/lib/python3.5/site-packages/cudf/dataframe/columnops.py in as_column(arbitrary, nan_as_null)
    261         else:
    262             if arbitrary.buffers()[0]:
--> 263                 pamask = Buffer(np.array(arbitrary.buffers()[0]))
    264             else:
    265                 pamask = None

/conda/envs/cudf/lib/python3.5/site-packages/cudf/dataframe/buffer.py in __init__(self, mem, size, capacity, categorical)
     33         if capacity is None:
     34             capacity = size
---> 35         self.mem = cudautils.to_device(mem)
     36         _BufferSentry(self.mem).ndim(1)
     37         self.size = size

/conda/envs/cudf/lib/python3.5/site-packages/cudf/utils/cudautils.py in to_device(ary)
     18 
     19 def to_device(ary):
---> 20     dary, _ = rmm.auto_device(ary)
     21     return dary
     22 

/conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi/wrapper.py in auto_device(self, obj, stream, copy)
    254 
    255             if copy:
--> 256                 devobj.copy_to_device(obj, stream=stream)
    257             return devobj, True
    258 

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devices.py in _require_cuda_context(*args, **kws)
    210     def _require_cuda_context(*args, **kws):
    211         get_context()
--> 212         return fn(*args, **kws)
    213 
    214     return _require_cuda_context

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devicearray.py in copy_to_device(self, ary, stream)
    203 
    204             sz = min(_driver.host_memory_size(ary), self.alloc_size)
--> 205             _driver.host_to_device(self, ary, sz, stream=stream)
    206 
    207     @devices.require_context

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/driver.py in host_to_device(dst, src, size, stream)
   1807         fn = driver.cuMemcpyHtoD
   1808 
-> 1809     fn(device_pointer(dst), host_pointer(src), size, *varargs)
   1810 
   1811 

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/driver.py in safe_cuda_api_call(*args)
    288             _logger.debug('call driver api: %s', libfn.__name__)
    289             retcode = libfn(*args)
--> 290             self._check_error(fname, retcode)
    291         return safe_cuda_api_call
    292 

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/driver.py in _check_error(self, fname, retcode)
    323                     _logger.critical(msg, _getpid(), self.pid)
    324                     raise CudaDriverError("CUDA initialized before forking")
--> 325             raise CudaAPIError(retcode, msg)
    326 
    327     def get_device(self, devnum=0):

CudaAPIError: [700] Call to cuMemcpyHtoD results in UNKNOWN_CUDA_ERROR
```

Once this error happens, all downstream cudf actions cause the same memory copy error.

If i restart my kernel and do the following, I get no error:

```
s = cudf.Series([1,2,3,4,4])
print(cudf.concat([s,s]))

       
 0    1
 1    2
 2    3
 3    4
 4    4
 0    1
 1    2
 2    3
 3    4
 4    4
```
CUDA 9.2, 410.48, V100, Ubuntu 16.04. Cudf 0.3 installed with defaults via `conda`.

**Comment by aguinaldoabbj** [comment]
After a bunch of errors regarding pyarrow (I had to convert my whole dataframe to string), I'm getting the exact same "[700] Call to cuMemcpyHtoD results in UNKNOWN_CUDA_ERROR" error when trying to convert a pandas dataframe do cudf (cudf.Dataframe.from_pandas(df)).

**Comment by jrhemstad** [comment]
@dantegd @harrism I assume this has something to do with the new `concat` libcudf function?

**Comment by dantegd** [comment]
At least the original log seems to fail in `s = cudf.Series([1,2,3,None,4])`, it doesn’t even get to the concat. It seems related to using PyArrow with the array containing a `None` since it fails in `data = as_column(pa.array(arbitrary))` which we use to create the series from the array. Will take a look on Monday, but @kkraus14 might have insight on this.

**Comment by dantegd** [comment]
Also as far as I remember the new concat does support the valid bitmask in both c++ and python, but it was a while ago I looked at it.

**Comment by kkraus14** [comment]
I believe this was fixed in #427, is this not the case?

**Comment by beckernick** [comment]
@kkraus14 You're right. This issue is now resolved with the conda release of cudf 0.4. Closing this issue. @aguinaldoabbj , if you are still running into memory copy issues when using `from_pandas` please create a new issue with steps to reproduce the error. Thanks!
